src/test/fuzz/util.cpp

```diff
@@ -524,7 +525,23 @@ CNetAddr ConsumeNetAddr(FuzzedDataProvider& fuzzed_data_provider) noexcept
     } else if (network == Network::NET_INTERNAL) {
         net_addr.SetInternal(fuzzed_data_provider.ConsumeBytesAsString(32));
     } else if (network == Network::NET_ONION) {
-        net_addr.SetSpecial(fuzzed_data_provider.ConsumeBytesAsString(32));
+        auto pub_key{fuzzed_data_provider.ConsumeBytes<uint8_t>(torv3::ADDR_SIZE)};
+        pub_key.resize(torv3::ADDR_SIZE);
+        // Create valid onion address to prevent the address checksum from
+        // being a fuzz blocker.
+        uint8_t check_sum[torv3::CHECKSUM_LEN];
+        torv3::Checksum({pub_key.data(), pub_key.size()}, check_sum);
```

Comment:
I had same question from @MarcoFalke, seems similar to`OnionToString`:
```cpp
std::string OnionToString(Span<const uint8_t> addr)
{
    uint8_t checksum[torv3::CHECKSUM_LEN];
    torv3::Checksum(addr, checksum);
    // TORv3 onion_address = base32(PUBKEY | CHECKSUM | VERSION) + ".onion"
    prevector<torv3::TOTAL_LEN, uint8_t> address{addr.begin(), addr.end()};
    address.insert(address.end(), checksum, checksum + torv3::CHECKSUM_LEN);
    address.insert(address.end(), torv3::VERSION, torv3::VERSION + sizeof(torv3::VERSION));
    return EncodeBase32(address) + ".onion";
}
```

Reply:
Oh, well, I was unaware of `OnionToString()`. The newly added code is a duplicate of that. @jonatack has done a good job isolating it. Now it can be reused, cool!

Reply:
Was also unaware of `OnionToString`. Looks like the better approach here would be to make that public in `netaddress.h` instead of moving the torv3 utils like i am doing here? Let me know if y'all prefer that (thumbs up/thumbs down).